### PR TITLE
fixed timestamps input&output not synchronized 

### PR DESCRIPTION
**Pull request by sirawats**
## Basic Info

| Info | Please fill out this column |
| ------ |------|
| Ticket | #376 
| Test on | Ubuntu 20.04 ROS2 Foxy, rosbag 

---

## Description of contribution in a few bullet points

*  I changed msg.header.stamp in `SlamToolbox::updateMap` and `SlamToolbox::publishTransformLoop` from this->now() to latest `/scan` timestamps (store in ::laserCallback)  
 
## Description of documentation updates required from your changes

---

### CODE REVIEW

File: include/slam_toolbox/visualization_utils.hpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ inline visualization_msgs::msg::Marker toMarker(
   marker.color.b = 0.0;
   marker.color.a = 1.;
   marker.action = visualization_msgs::msg::Marker::ADD;
-  marker.lifetime = rclcpp::Duration::from_nanoseconds(0);
+  marker.lifetime = rclcpp::Duration(0.);
```
Comment (SteveMacenski) [inline include/slam_toolbox/visualization_utils.hpp]: Return back to `from_nanoseconds`

File: src/slam_toolbox_common.cpp [hunk 2]
```diff
@@ -231,7 +231,7 @@ void SlamToolbox::publishTransformLoop(
       msg.transform = tf2::toMsg(map_to_odom_);
       msg.child_frame_id = odom_frame_;
       msg.header.frame_id = map_frame_;
-      msg.header.stamp = this->now() + transform_timeout_;
+      msg.header.stamp = scan_timestamped + transform_timeout_; // 8 Apr.21 | remove this->now
```
Comment (SteveMacenski) [inline src/slam_toolbox_common.cpp]: No need for the comment, git history will have this.

**Review by SteveMacenski — COMMENTED** [review]
A couple of small changes and LGTM

**Comment by sirawats** [comment]
Thanks!, resolved

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
LGTM, but please run the linters, I'm seeing several violations

**Comment by sirawats** [comment]
You mean ament_cpplint right? I removed those extra spaces. Thanks for pointing out.

**Review by SteveMacenski — APPROVED** [review]

**Comment by SteveMacenski** [comment]
Thanks!

**Pull request merged**